### DISCUSSION TITLE
VRS 2.0.0 Community Review Ballot

### DISCUSSION THREAD

**Discussion by ahwagner** [ga4gh/vrs#605] [General]
This is the ballot pre-release of the VRS 2.0 specification, open for community review following the GA4GH Product Development and Approval process.

## Important Links
**Documentation**: https://vrs.ga4gh.org/en/2.0.0-ballot.2024-11/
**Release log**: https://vrs.ga4gh.org/en/2.0.0-ballot.2024-11/releases/2.0.html
**Schemas**: https://github.com/ga4gh/vrs/tree/2.0.0-ballot.2024-11.1/schema/vrs/json
**Discussion**: this page!

**Full Changelog**: https://github.com/ga4gh/vrs/compare/1.3.0...2.0.0-ballot.2024-11.1

<hr /><em>This discussion was created from the release <a href='https://github.com/ga4gh/vrs/releases/tag/2.0.0-ballot.2024-11.1'>VRS 2.0.0 Community Review Ballot</a>.</em>

**Comment by korikuzma** [comment]
VRS Roadmap in https://vrs.ga4gh.org/en/2.0.0-ballot.2024-11/releases/index.html gives 404. Should it point to [project 12](https://github.com/orgs/ga4gh/projects/12) instead?

**Reply by monicacecilia** [reply]
This was fixed.

**Comment by korikuzma** [comment]
https://vrs.ga4gh.org/en/2.0.0-ballot.2024-11/appendices/class_diagram.html

1. I believe `vrs` in `sequenceString` and `residue` should be blue
2. Is it necessary to include Normative in the legend if no classes in the diagram have this maturity level?
3. `SequenceExpression` inherits from Entity, but this is not expressed in the diagram. Is there a reason for this?

**Reply by monicacecilia** [reply]
Points 1. and 2. were addressed. I don't see an update for point 3.

**Comment by korikuzma** [comment]
`CopyNumberCount` and `CopyNumberChange` no longer inherit from `CopyNumber`. However, the RTD does not reflect this change to inherit from `SystemicVariation` directly.

![Screenshot 2024-11-26 at 16 54 10](https://github.com/user-attachments/assets/627182e0-5e52-4b9e-b249-86e8df68c17c)

![Screenshot 2024-11-26 at 16 52 15](https://github.com/user-attachments/assets/88d6eb62-12eb-494a-baee-b9cd6876045d)

Additionally, it would be nice to have the organization follow the other classes, where Definition and Information Model and Examples are in the expanded section.

**Reply by monicacecilia** [reply]
Has not been fixed.  👀

**Reply by korikuzma** [reply]
I have PR out for some of these! #623

**Comment by korikuzma** [comment]
The [Primitive Types section](https://vrs.ga4gh.org/en/2.0.0-ballot.2024-11/concepts/AdditionalDataTypes/index.html#primitive-types) includes `Range`, which is not correct.

![Screenshot 2024-11-26 at 17 04 47](https://github.com/user-attachments/assets/6b8838fd-889f-4960-a877-172555c0d6d6)

The class diagram correctly puts `Range` in the General Purpose Data Classes

**Comment by korikuzma** [comment]
https://vrs.ga4gh.org/en/2.0.0-ballot.2024-11/appendices/ga4gh_identifiers.html#ga4gh-digest-keys

![Screenshot 2024-11-26 at 17 12 42](https://github.com/user-attachments/assets/86a536b4-f47a-4dd2-9988-e51859878a1a)

I'm not sure the thought is complete. 

Also, this is minor and just an example, but the brackets not being closed bothers me.

**Reply by monicacecilia** [reply]
Just here to let you know that said bracket offense hasn't been fixed. 😉 ... and in all seriousness, that the thought still appears incomplete.

**Comment by korikuzma** [comment]
<img width="1106" alt="Screenshot 2024-11-26 at 23 37 45" src="https://github.com/user-attachments/assets/44c61419-ecd1-420b-94af-d47b18fb55ff">

GKS Core references are not working because it's using targets that aren't defined, such as

```
:ref:`gks-core:Entity`
```

when it should be using (no `gks-core:` prefix)

```
:ref:`Entity`
```

When changing the reference, this links correctly to https://vrs.ga4gh.org/en/2.0.0-ballot.2024-11/concepts/AdditionalDataTypes/Element.html

By the way, I'm not sure if I should be creating issues for these. We can convert to issues if leads agree.

**Comment by korikuzma** [comment]
Currently `Location` is an ABC, but `SequenceLocation` is currently the only kind of `Location`. [This](https://github.com/ga4gh/vrs/issues/561#issuecomment-2502155856) comment leads me to believe it will be removed in the future. Is there a reason for this, such as potentially adding more types of Locations in the future? From a dev standpoint, this feels like a potential case for YAGNI

 Given that the `CopyNumber` ABC was removed and `CopyNumberCount` and `CopyNumberChange` directly inherit from `SystemicVariation` (as seen in #600), I would assume that we'd have `SequenceLocation` directly inherit from `Ga4ghIdentifiableObject` too. But I might be missing why CN was removed.

**Comment by andrewpatto** [comment]
![Screenshot 2024-11-28 at 9 00 15 am](https://github.com/user-attachments/assets/bc539132-6c4d-4ba2-8c08-f95208d4f677)

The intro paragraph for Systemic Variation is identical to the intro paragraph of Molecular Variation - which is probably an editing oversight

**Reply by korikuzma** [reply]
@ahwagner @larrybabb This should probably get resolved before 2.0 product review, right?

**Comment by korikuzma** [comment]
`code` is missing in imported classes: https://vrs.ga4gh.org/en/2.0.0-ballot.2024-11/concepts/AdditionalDataTypes/index.html#imported-classes 

This can be fixed by fixing the case of code RST file (or references). It'll probably be good to have this updated in gks-core with proper lowercase now (since iriReference files have correct case)
